hydropandas/io/lizard.py

```diff
@@ -0,0 +1,689 @@
+        combine: keeps hand and diver measurements separeted
+        The default is merge.
+    only_metadata : bool, optional
+        if True only metadata is returned and no time series data. The
+        default is False.
+
+
+    Returns
+    -------
+    obs_list
+        list of observations
+
+    """
+
+    if isinstance(codes, str):
+        codes = [codes]
+
+    if not hasattr(codes, "__iter__"):
+        raise TypeError("argument 'codes' should be an iterable")
+
+    obs_list = []
+    for code in codes:
+        groundwaterstation_metadata = get_metadata_mw_from_code(code)
+        if tube_nr == "all":
+            for metadata_tube in groundwaterstation_metadata["filters"]:
+                tube_nr = int(metadata_tube["code"][-3:])
+                o = ObsClass.from_lizard(
+                    code,
+                    tube_nr,
+                    tmin,
+                    tmax,
+                    type_timeseries,
+                    only_metadata=only_metadata,
+                )
+                obs_list.append(o)
+        else:
+            o = ObsClass.from_lizard(
+                code, tube_nr, tmin, tmax, type_timeseries, only_metadata=only_metadata
+            )
+            obs_list.append(o)
+
+    return obs_list
+
+
+def get_obs_list_from_extent(
+    extent,
+    ObsClass,
+    tube_nr="all",
+    tmin=None,
+    tmax=None,
+    type_timeseries="merge",
+    only_metadata=False,
+    page_size=100,
+    nr_threads=10,
+):
+    """
+    get all observations within a specified extent
+    Parameters
+    ----------
+    extent : list or a shapefile
+        get groundwater monitoring wells wihtin this extent [xmin, xmax, ymin, ymax]
+        or within a predefined Polygon from a shapefile
+    ObsClass : type
+        class of the observations, e.g. GroundwaterObs
+    tube_nr : lst of str
+        list of tube numbers of the monitoring wells that should be selected.
+        By default 'all' available tubes are selected.
+    tmin : str YYYY-m-d, optional
+        start of the observations, by default the entire serie is returned
+    tmax : Ttr YYYY-m-d, optional
+        end of the observations, by default the entire serie is returned
+    type_timeseries : str, optional
+        merge: the hand and diver measurements into one time series (merge; default) or
+        combine: keeps hand and diver measurements separeted
+        The default is merge.
+    only_metadata : bool, optional
+        if True only metadata is returned and no time series data. The
+        default is False.
+
+
+    Returns
+    -------
+    obs_col : TYPE
+        ObsCollection DataFrame with the 'obs' column
+
+    """
+
+    if isinstance(extent, (list, tuple)):
+        polygon_T = extent_to_wgs84_polygon(extent)
+
+    elif isinstance(extent, str) or isinstance(extent, pathlib.PurePath):
+        polygon = geopandas.read_file(extent)
+        polygon_T = polygon.to_crs("WGS84", "EPSG:28992").loc[0, "geometry"]
```

Comment:
Is this transformation correct, given some of the issues with EPSG 28992 we've encountered before?